pkg/occlient/occlient.go

```diff
@@ -1306,7 +1313,59 @@ func (c *Client) DeleteProject(name string) error {
 	if err != nil {
 		return errors.Wrap(err, "unable to delete project")
 	}
-	return nil
+
+	// wait for delete to complete
+	w, err := c.projectClient.Projects().Watch(metav1.ListOptions{
+		FieldSelector: fields.Set{"metadata.name": name}.AsSelector().String(),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "unable to watch project")
+	}
+
+	defer w.Stop()
+	for {
+		val, ok := <-w.ResultChan()
+		// When marked for deletion... val looks like:
+		/*
+			val: {
+				Type:MODIFIED
+				Object:&Project{
+					ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{...},
+					Spec:ProjectSpec{...},
+					Status:ProjectStatus{
+						Phase:Terminating,
+					},
+				}
+			}
+		*/
+		// Post deletion val will look like:
+		/*
+			val: {
+				Type:DELETED
+				Object:&Project{
+					ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{...},
+					Spec:ProjectSpec{...},
+					Status:ProjectStatus{
+						Phase:,
+					},
+				}
+			}
+		*/
+		if !ok {
+			return fmt.Errorf("Received unexpected signal %+v on project watch channel", val)
+		}
+		// So we depend on val.Type as val.Object.Status.Phase is just empty string and not a mapped value constant
+		if prj, ok := val.Object.(*projectv1.Project); ok {
+			glog.V(4).Infof("Status of delete of project %s is %s", name, prj.Status.Phase)
+			switch prj.Status.Phase {
+			//prj.Status.Phase can only be "Terminating" or "Active" or ""
+			case "":
+				if val.Type == watch.Deleted {
```

Comment:
You should also check for `Error` type to be sure that you won't end up in the endless loop in case of error.

Reply:
Ack